### PR TITLE
Add test_broken for #24333

### PR DESCRIPTION
**Pull request by oscardssmith**
https://github.com/JuliaLang/julia/issues/24333

### CODE REVIEW

File: test/subtype.jl [hunk 1]
```diff
@@ -1811,3 +1811,6 @@ end
 @testintersect(Type{T} where T>:Missing,
                Type{Some{T}} where T,
                Union{})
+
+# issue #24333
+@test !(Type{Union{Ref,Cvoid}} <: Type{Union{T,Cvoid}} where T)
```
Comment (dkarrasch) [inline test/subtype.jl:1816]: I thought `Type{Union{Ref,Cvoid}} <: Type{Union{T,Cvoid}} where T` was supposed to be `true`, according to the issue?

**Review by dkarrasch — CHANGES_REQUESTED** [review]

File: test/subtype.jl (hunk 1 above)
Comment (oscardssmith) [reply test/subtype.jl:1816]: Oh. I totally misread that issue. Should this be `@test_broken`

Comment (JeffBezanson) [reply test/subtype.jl:1816]: Yes it's supposed to be true.

**Comment by oscardssmith** [comment]
Failed build was a timeout. Merge?

**Comment by dkarrasch** [comment]
> Merge?

Why not, but that doesn't close the issue, of course. I'm not aware of the fact that we add "broken tests" for open issues, though. Typically, we add such tests when the issues are fixed, to avoid regression, not beforehand as a reminder of the issue.

**Comment by JeffBezanson** [comment]
No harm in adding it I suppose.

**Pull request merged**